pkg/pod/entrypoint.go

```diff
@@ -109,6 +110,18 @@ func orderContainers(entrypointImage string, steps []corev1.Container) (corev1.C
 				"-post_file", filepath.Join(mountPoint, fmt.Sprintf("%d", i)),
 				"-termination_path", terminationPath,
 			}
+			if results != nil && len(steps) == 1 {
```

Comment:
Maybe wrap the logic to append the results command line flags in a function and reuse?

Reply:
I'd like to propose that the results arg is always appended to step containers if results != nil.  This way we don't need to worry about the distinction between `1` and `len - 1` cases.

This will mean that every step's entrypointer will search for results and write them to their termination log.  It will then be the TaskRun controller's responsibility to aggregate the results of all containers into the TaskRun Results field, performing any validation (such as checking whether a result is missing) as needed.